bin/report_summaries.py

```diff
@@ -291,20 +304,21 @@ def to_file(self):
 
 
 
-def main_(args_in):
+def main(args_in):
     default_samp_suffix = "_flat_sample.json"
     parser = argparse.ArgumentParser("Table Summary")
     parser.add_argument("-f", "--file-in", help="Path to the mikrokondo json summary")
     parser.add_argument("-s", "--sample-tag", help="Optional suffix and extension to name output samples.", default=default_samp_suffix)
     parser.add_argument("-o", "--out-file", help="output name plus the .tsv extension e.g. prefix.tsv")
+    parser.add_argument("-x", "--inx-id-token", help="A token to insert into the flattened json file names for separation of the irida next sample id.")
```

Comment:
I am curious if, rather than inserting the IRIDA Next id into the flattened JSON file names, it can be used to create a folder with contents being the flattened JSON report? That way, you don't have to worry about inserting tokens into a filename and then parsing them out from the string later on. You can then also insert the sample name as part of the flattened report file name.

That is name output files like:

```
FlattenedReports/IRIDA_NEXT_ID/SAMPLE_NAME.flat_sample.json
```

Then, you can iterate over all subdirectories in `FlattenedReports/`, and parse the IRIDA Next identifier from the sub-directory. That is in:

https://github.com/phac-nml/mikrokondo/blob/c036fb5025a7cfc4ced6b34227dd601cacf57620/main.nf#L113-L119

Pull out the IRIDA Next id (`external_id`) from the directory name instead of from the file name.

Reply:
good idea!! I think I will do that, that is much cleaner.

Reply:
fixed in: https://github.com/phac-nml/mikrokondo/pull/140/commits/39c8505824de4d5dbd577b7674858dc121103ce3

The output directory structure did not change, just the outputs from the script are structured.